### PR TITLE
web3 formatters correction

### PR DESCRIPTION
**Pull request by bas-vk**
This corrects input/output formatters which have become invalid.

### CODE REVIEW

File: rpc/api/admin_js.go [hunk 1]
```diff
@@ -9,73 +9,62 @@ web3._extend({
...
-			outputFormatter: function(obj) { return obj; }
+			inputFormatter: [null]
 		}),
 		new web3._extend.Method({
 			name: 'importChain',
 			call: 'admin_importChain',
 			params: 1,
-			inputFormatter: [web3._extend.utils.formatInputString],
-			outputFormatter: function(obj) { return obj; }
+			inputFormatter: [null]
```
Comment (frozeman) [inline rpc/api/admin_js.go]: What you put in here?

File: rpc/api/admin_js.go [hunk 2]
```diff
@@ -9,73 +9,62 @@ web3._extend({
...
 		}),
 		new web3._extend.Method({
 			name: 'verbosity',
 			call: 'admin_verbosity',
 			params: 1,
-			inputFormatter: [web3._extend.utils.formatInputInt],
-			outputFormatter: web3._extend.formatters.formatOutputBool
+			inputFormatter: [web3._extend.utils.toDecimal]
```
Comment (frozeman) [inline rpc/api/admin_js.go]: Thats probably `fromDecimal`, because you transform from decimal to hex

File: rpc/api/admin_js.go [hunk 3]
```diff
@@ -9,73 +9,62 @@ web3._extend({
...
 		}),
 		new web3._extend.Method({
 			name: 'setSolc',
 			call: 'admin_setSolc',
 			params: 1,
-			inputFormatter: [web3._extend.utils.formatInputString],
-			outputFormatter: web3._extend.formatters.formatOutputString
+			inputFormatter: [null]
```
Comment (frozeman) [inline rpc/api/admin_js.go]: This should probably be `fromAscii`. As strings are transported using Hex according to the specs.

File: rpc/api/admin_js.go [hunk 4]
```diff
@@ -9,73 +9,62 @@ web3._extend({
...
 		}),
 		new web3._extend.Method({
 			name: 'startRPC',
 			call: 'admin_startRPC',
 			params: 4,
-			inputFormatter: [web3._extend.utils.formatInputString,web3._extend.utils.formatInputInteger,web3._extend.utils.formatInputString,web3._extend.utils.formatInputString],
-			outputFormatter: web3._extend.formatters.formatOutputBool
+			inputFormatter: [null, web3._extend.utils.toDecimal, null, null]
```
Comment (frozeman) [inline rpc/api/admin_js.go]: `web3._extend.utils.fromDecimal`. The others may use `fromAscii` i guess

File: rpc/api/debug_js.go [hunk 5]
```diff
@@ -9,50 +9,43 @@ web3._extend({
 			name: 'printBlock',
 			call: 'debug_printBlock',
 			params: 1,
-			inputFormatter: [web3._extend.formatters.formatInputInt],
-			outputFormatter: web3._extend.formatters.formatOutputString
+			inputFormatter: [web3._extend.formatters.inputBlockNumberFormatter]
```
Comment (frozeman) [inline rpc/api/debug_js.go]: Do you put a block into this method?

Or did you meant to use `outputBlockFormatter`?

File: rpc/api/debug_js.go [hunk 6]
```diff
@@ -9,50 +9,43 @@ web3._extend({
...
 		}),
 		new web3._extend.Method({
 			name: 'getBlockRlp',
 			call: 'debug_getBlockRlp',
 			params: 1,
-			inputFormatter: [web3._extend.formatters.formatInputInt],
-			outputFormatter: web3._extend.formatters.formatOutputString
+			inputFormatter: [web3._extend.formatters.inputBlockNumberFormatter]
```
Comment (frozeman) [inline rpc/api/debug_js.go]: Same here (?)

File: rpc/api/debug_js.go [hunk 7]
```diff
@@ -9,50 +9,43 @@ web3._extend({
...
 		}),
 		new web3._extend.Method({
 			name: 'setHead',
 			call: 'debug_setHead',
 			params: 1,
-			inputFormatter: [web3._extend.formatters.formatInputInt],
-			outputFormatter: web3._extend.formatters.formatOutputBool
+			inputFormatter: [web3._extend.formatters.inputBlockNumberFormatter]
```
Comment (frozeman) [inline rpc/api/debug_js.go]: Same here (?)

File: rpc/api/debug_js.go [hunk 8]
```diff
@@ -9,50 +9,43 @@ web3._extend({
...
 		}),
 		new web3._extend.Method({
 			name: 'seedHash',
 			call: 'debug_seedHash',
 			params: 1,
-			inputFormatter: [web3._extend.formatters.formatInputInt],
-			outputFormatter: web3._extend.formatters.formatOutputString
-		})		,
+			inputFormatter: [web3._extend.formatters.inputBlockNumberFormatter]
```
Comment (frozeman) [inline rpc/api/debug_js.go]: Same here (?)

File: rpc/api/debug_js.go [hunk 9]
```diff
@@ -9,50 +9,43 @@ web3._extend({
...
+		}),
 		new web3._extend.Method({
 			name: 'dumpBlock',
 			call: 'debug_dumpBlock',
 			params: 1,
-			inputFormatter: [web3._extend.formatters.formatInputInt],
-			outputFormatter: function(obj) { return obj; }
+			inputFormatter: [web3._extend.formatters.inputBlockNumberFormatter]
```
Comment (frozeman) [inline rpc/api/debug_js.go]: Same here (?)

File: rpc/api/debug_js.go [hunk 10]
```diff
@@ -9,50 +9,43 @@ web3._extend({
...
 		}),
 		new web3._extend.Method({
 			name: 'metrics',
 			call: 'debug_metrics',
 			params: 1,
-			inputFormatter: [web3._extend.formatters.formatInputBool],
-			outputFormatter: function(obj) { return obj; }
+			inputFormatter: [null]
```
Comment (frozeman) [inline rpc/api/debug_js.go]: Whats the parameter here?

File: rpc/api/eth_js.go [hunk 11]
```diff
@@ -12,23 +12,20 @@ web3._extend({
...
-			outputFormatter: web3._extend.formatters.formatOutputString
+			inputFormatter: [web3._extend.utils.toAddress, null]
 		}),
 		new web3._extend.Method({
 			name: 'resend',
 			call: 'eth_resend',
 			params: 3,
-			inputFormatter: [function(obj) { return obj; },web3._extend.formatters.formatInputString,web3._extend.formatters.formatInputString],
-			outputFormatter: web3._extend.formatters.formatOutputString
+			inputFormatter: [null, null, null]
```
Comment (frozeman) [inline rpc/api/eth_js.go]: This seems it needs inout formatting

File: rpc/api/miner_js.go [hunk 12]
```diff
@@ -9,50 +9,43 @@ web3._extend({
...
-			outputFormatter: web3._extend.formatters.formatOutputBool
+			inputFormatter: [null]
 		}),
 		new web3._extend.Method({
 			name: 'stop',
 			call: 'miner_stop',
 			params: 1,
-			inputFormatter: [web3._extend.formatters.formatInputInt],
-			outputFormatter: web3._extend.formatters.formatOutputBool
+			inputFormatter: [null]
```
Comment (frozeman) [inline rpc/api/miner_js.go]: This has a parameter?

File: rpc/api/miner_js.go [hunk 13]
```diff
@@ -9,50 +9,43 @@ web3._extend({
...
 		}),
 		new web3._extend.Method({
 			name: 'setExtra',
 			call: 'miner_setExtra',
 			params: 1,
-			inputFormatter: [web3._extend.utils.formatInputString],
-			outputFormatter: web3._extend.formatters.formatOutputBool
+			inputFormatter: [null]
```
Comment (frozeman) [inline rpc/api/miner_js.go]: This probably needs `fromDecimal`

File: rpc/api/miner_js.go [hunk 14]
```diff
@@ -9,50 +9,43 @@ web3._extend({
...
 		}),
 		new web3._extend.Method({
 			name: 'setGasPrice',
 			call: 'miner_setGasPrice',
 			params: 1,
-			inputFormatter: [web3._extend.utils.formatInputString],
-			outputFormatter: web3._extend.formatters.formatOutputBool
+			inputFormatter: [null]
```
Comment (frozeman) [inline rpc/api/miner_js.go]: Same here

File: rpc/api/miner_js.go [hunk 15]
```diff
@@ -9,50 +9,43 @@ web3._extend({
...
-			outputFormatter: web3._extend.formatters.formatOutputBool
+			inputFormatter: []
 		}),
 		new web3._extend.Method({
 			name: 'makeDAG',
 			call: 'miner_makeDAG',
 			params: 1,
-			inputFormatter: [web3._extend.formatters.inputDefaultBlockNumberFormatter],
-			outputFormatter: web3._extend.formatters.formatOutputBool
+			inputFormatter: [web3._extend.formatters.inputDefaultBlockNumberFormatter]
```
Comment (frozeman) [inline rpc/api/miner_js.go]: Is that correct?

File: rpc/api/net_js.go [hunk 16]
```diff
@@ -2,45 +2,25 @@ package api
...
 			name: 'peers',
-			getter: 'net_peers',
-			outputFormatter: function(obj) { return obj; }
+			getter: 'net_peers'
 		}),
 		new web3._extend.Property({
 			name: 'version',
-			getter: 'net_version',
-			outputFormatter: web3._extend.formatters.formatOutputString
+			getter: 'net_version'
```
Comment (frozeman) [inline rpc/api/net_js.go]: This should already exist. its in `web3.version.network`

File: rpc/api/personal_js.go [hunk 17]
```diff
@@ -9,23 +9,21 @@ web3._extend({
 			name: 'newAccount',
 			call: 'personal_newAccount',
 			params: 1,
-			inputFormatter: [web3._extend.formatters.formatInputString],
-			outputFormatter: web3._extend.formatters.formatOutputString
+			inputFormatter: [null],
```
Comment (frozeman) [inline rpc/api/personal_js.go]: Should probably use `fromAscii` to send the pw in hex.

File: rpc/api/personal_js.go [hunk 18]
```diff
@@ -9,23 +9,21 @@ web3._extend({
...
+			outputFormatter: web3._extend.utils.toAddress
 		}),
 		new web3._extend.Method({
 			name: 'unlockAccount',
 			call: 'personal_unlockAccount',
 			params: 3,
-			inputFormatter: [web3._extend.formatters.formatInputString,web3._extend.formatters.formatInputString,web3._extend.formatters.formatInputInt],
-			outputFormatter: web3._extend.formatters.formatOutputBool
+			inputFormatter: [null, null, null]
```
Comment (frozeman) [inline rpc/api/personal_js.go]: same here

File: rpc/api/utils.go [hunk 19]
```diff
@@ -101,6 +91,8 @@ var (
 		"net": []string{
 			"peerCount",
 			"listening",
+			"version",
```
Comment (frozeman) [inline rpc/api/utils.go]: Thats wrong

File: rpc/api/eth_js.go (hunk 11 above)
Comment (frozeman) [inline rpc/api/eth_js.go]: `formatters.inputTransactionFormatter`, `utils.fromDecimal`, `utils.fromDecimal`

File: rpc/api/miner_js.go (hunk 12 above)
Comment (frozeman) [inline rpc/api/miner_js.go]: `utils.fromDecimal`

File: rpc/api/admin_js.go (hunk 1 above)
Comment (bas-vk) [inline rpc/api/admin_js.go]: A filename path, as discussed this isn't changed.

File: rpc/api/admin_js.go (hunk 3 above)
Comment (bas-vk) [inline rpc/api/admin_js.go]: A filename path, as discussed this isn't changed.

File: rpc/api/admin_js.go (hunk 4 above)
Comment (bas-vk) [inline rpc/api/admin_js.go]: Second argument is the portnumber which is an integer. Removed the formatter.

File: rpc/api/debug_js.go (hunk 5 above)
Comment (bas-vk) [inline rpc/api/debug_js.go]: As input it expects the blocknumber or one of the strings ['earliest', 'latest', 'pending']

File: rpc/api/debug_js.go (hunk 10 above)
Comment (bas-vk) [inline rpc/api/debug_js.go]: A boolean

File: rpc/api/miner_js.go (hunk 12 above)
Comment (bas-vk) [inline rpc/api/miner_js.go]: An integer specifying the number of threads

File: rpc/api/miner_js.go (hunk 13 above)
Comment (bas-vk) [inline rpc/api/miner_js.go]: This is a string which will be included in blocks that this node mines.

File: rpc/api/miner_js.go (hunk 15 above)
Comment (bas-vk) [inline rpc/api/miner_js.go]: Yes, it expects a block number or one of the strings ['earliest', 'latest', 'pending']

File: rpc/api/net_js.go (hunk 16 above)
Comment (bas-vk) [inline rpc/api/net_js.go]: deleted

File: rpc/api/personal_js.go (hunk 17 above)
Comment (bas-vk) [inline rpc/api/personal_js.go]: No conversion to hex is done as discussed.

File: rpc/api/utils.go (hunk 19 above)
Comment (bas-vk) [inline rpc/api/utils.go]: removed

**Comment by frozeman** [comment]
Seems fine now.

**Comment by obscuren** [comment]
Need rebase and somehow contains other people's commits

**Pull request closed without merging**